subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6548,6 +6549,18 @@ static void le_per_adv_sync_report(struct pdu_data *pdu_data,
 		return;
 	}
 
+	/* FIXME: make node_rx self sufficient with data when generating so that
+	 *        sync context is not accessed in HCI, the context could be
+	 *        released under terminate or reset scenarios.
+	 */
+	sync = HDR_LLL2ULL(ftr->param);
+	if (!sync->timeout_reload) {
+		/* NOTE: This is a workaround, to use the field in the context
```

Comment:
I know this is a workaround.... but it is a bit wired. This is conceptually like using a malloc-ed memory after call to free.
Should it be done this way?